### PR TITLE
Updates to mongo module, adding tests

### LINKED ISSUE #62900: [BUG] MongoDB module not up to date with latest pymongo library
**Description**
MongoDB module not up to date with latest pymongo library

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Updating various mongodb functions to sync up with latest version ofPymongo.  Adding some unit tests for mongodb module.

### What issues does this PR fix or reference?
Fixes: #62900

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/mongodb.py [hunk 1]
```diff
@@ -297,12 +297,14 @@ def user_create(
     if not roles:
         roles = []
 
+    _roles = [{"role": _role, "db": database} for _role in roles]
     try:
         log.info("Creating user %s", name)
         mdb = pymongo.database.Database(conn, database)
-        mdb.add_user(name, passwd, roles=roles)
+        ret = mdb.command("createUser", name, pwd=passwd, roles=_roles)
```
Comment (Ch3LL) [inline salt/modules/mongodb.py:304]: why are you adding this call to a variable, when the variables is not being used.

This also applies to line 331 and 406

File: salt/modules/mongodb.py [hunk 2]
```diff
@@ -326,7 +328,7 @@ def user_remove(
     try:
         log.info("Removing user %s", name)
         mdb = pymongo.database.Database(conn, database)
-        mdb.remove_user(name)
+        ret = mdb.command("dropUser", name)
     except pymongo.errors.PyMongoError as err:
         log.error("Creating database %s failed with error: %s", name, err)
```
Comment (Ch3LL) [inline salt/modules/mongodb.py:333]:
```suggestion
        log.error("Removing user %s failed with error: %s", name, err)
```

File: salt/modules/mongodb.py [hunk 3]
```diff
@@ -449,6 +451,107 @@ def user_revoke_roles(
...
+    authdb=None,
+):
+    """
+    Create a collection in the specified database.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mongodb.create_collection mycollection <user> <password> <host> <port> <database>
```
Comment (Ch3LL) [inline salt/modules/mongodb.py:470]: versionadded 3006

File: salt/modules/mongodb.py [hunk 4]
```diff
@@ -449,6 +451,107 @@ def user_revoke_roles(
...
+
```
Comment (Ch3LL) [inline salt/modules/mongodb.py:471]: Can you add documentation for each of the args/kwargs

File: salt/modules/mongodb.py [hunk 5]
```diff
@@ -449,6 +451,107 @@ def user_revoke_roles(
...
+):
+    """
+    Drop a collection in the specified database.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mongodb.collection_drop mycollection <user> <password> <host> <port> <database>
+
```
Comment (Ch3LL) [inline salt/modules/mongodb.py:506]: versionadded 3006

and can you add docs for each arg/kwarg here.

File: salt/modules/mongodb.py [hunk 6]
```diff
@@ -449,6 +451,107 @@ def user_revoke_roles(
...
+    authdb=None,
+):
+    """
+    List the collections available in the specified database.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mongodb.collections_list mycollection <user> <password> <host> <port> <database>
```
Comment (Ch3LL) [inline salt/modules/mongodb.py:539]: same here

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this also requires a changelog

File: salt/modules/mongodb.py (hunk 1 above)
Comment (garethgreenaway) [reply salt/modules/mongodb.py:454]: Something left over from testing.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**